### PR TITLE
Fix on B2C validation copyWithZone

### PR DESCRIPTION
**Pull request by unpluggedk**
implement copyWithZone separately for B2C authority, without format validation.

### CODE REVIEW

File: IdentityCore/src/validation/MSIDB2CAuthority.m [hunk 1]
```diff
@@ -165,4 +165,12 @@ + (NSURL *)normalizedAuthorityUrl:(NSURL *)url
     return [NSURL URLWithString:normalizedAuthorityUrl];
 }
 
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone
+{
+    MSIDB2CAuthority *authority = [[self.class allocWithZone:zone] initWithURL:_url validateFormat:NO context:nil error:nil];
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDB2CAuthority.m]: shouldn't we also copy openid connect endpoint and metadata?

Comment (unpluggedk) [reply IdentityCore/src/validation/MSIDB2CAuthority.m]: yeeeeeeeesssssss

File: IdentityCore/src/validation/MSIDB2CAuthority.m [hunk 2]
```diff
@@ -165,4 +165,17 @@ + (NSURL *)normalizedAuthorityUrl:(NSURL *)url
...
+    
+    authority.openIdConfigurationEndpoint = [_openIdConfigurationEndpoint copyWithZone:zone];
+    authority.metadata = self.metadata;
+    authority.url = [_url copyWithZone:zone];
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDB2CAuthority.m]: Does authority need to be set again? It's already passed in initializer.

File: IdentityCore/src/validation/MSIDB2CAuthority.m [hunk 3]
```diff
@@ -165,4 +165,17 @@ + (NSURL *)normalizedAuthorityUrl:(NSURL *)url
     return [NSURL URLWithString:normalizedAuthorityUrl];
 }
 
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone
+{
+    MSIDB2CAuthority *authority = [[self.class allocWithZone:zone] initWithURL:_url validateFormat:NO context:nil error:nil];
```
Comment (oldalton) [inline IdentityCore/src/validation/MSIDB2CAuthority.m]: Should we copy url that we pass?

File: IdentityCore/src/validation/MSIDB2CAuthority.m (hunk 2 above)
Comment (unpluggedk) [reply IdentityCore/src/validation/MSIDB2CAuthority.m]: you mean URL? nope, I think we can skip this.

Comment (unpluggedk) [reply IdentityCore/src/validation/MSIDB2CAuthority.m]: Agreed and removed.

File: IdentityCore/src/validation/MSIDB2CAuthority.m (hunk 3 above)
Comment (unpluggedk) [reply IdentityCore/src/validation/MSIDB2CAuthority.m]: It seems like its related to above feedback which is addressed.

Comment (oldalton) [reply IdentityCore/src/validation/MSIDB2CAuthority.m]: nope, it should be  MSIDB2CAuthority *authority = [[self.class allocWithZone:zone] initWithURL:[_url copyWithZone:zone] validateFormat:NO context:nil error:nil];

**Pull request merged**